source/common/network/dns_impl.cc

```diff
@@ -169,20 +169,17 @@ void DnsResolverImpl::PendingResolution::onAresGetAddrInfoCallback(int status, i
 
   if (completed_) {
     if (!cancelled_) {
-      // TODO(chaoqin-li1123): remove this exception catching by refactoring.
-      //  We can't add a main thread assertion here because both this code is reused by dns filter
-      //  and executed in both main thread and worker thread. Maybe split the code for filter and
-      //  main thread.
-      TRY_NEEDS_AUDIT { callback_(resolution_status, std::move(address_list)); }
-      catch (const EnvoyException& e) {
+      // Use a raw try here because it is used in both main thread and filter.
+      // Can not convert to use status code as there may be unexpected exceptions.
```

Comment:
Add that this is mainly an issue in server_test.cc, and we should try to resolve these.